### PR TITLE
Channel members

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
Adds ability to view members in channel from channel info screen.
Adds reusable member_list component to display members

#### Ticket Link
#60 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] All new/modified APIs include changes to the drivers
- [X] Has UI changes
- [X] Includes text changes and localization file updates

### CODE REVIEW

File: app/navigation/routes.js [hunk 1]
```diff
@@ -16,6 +16,10 @@ export const Routes = {
     ChannelDrawer: {
         key: 'ChannelDrawer'
     },
+    ChannelMembers: {
+        key: 'ChannelMembers',
+        title: {id: 'channel_header.manageMembers', defaultMessage: 'Manage Members'}
```
Comment (hmhealey) [inline app/navigation/routes.js]: Does this need a transition?

**Comment by jarredwitt** [comment]
@hmhealey ready for another look

File: app/components/member_list/index.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        const dataSource = sections ? this.state.dataSource.cloneWithRowsAndSections(this.createSections(members)) : this.state.dataSource.cloneWithRows(members);
+        this.setState({
+            dataSource
+        });
+    }
+
+    createSections = (data) => {
+        const sections = {};
+        data.forEach((d) => {
+            const name = (d.first_name && d.last_name) ? d.first_name : d.username;
```
Comment (hmhealey) [inline app/components/member_list/index.js]: A user's display name can also be their nickname, and is controlled by the current user's preferences making this a bit more complicated. You'll want to use the (poorly named) `displayUsername` function available in `user_utils.js` for this.

File: app/components/member_list/index.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        return (
+            <View style={style.sectionContainer}>
+                <Text style={style.sectionText}>{sectionId}</Text>
+            </View>
+        );
+    }
+
+    renderRow = (data) => {
+        const {id, first_name: firstName, last_name: lastName, username, status} = data;
+        const displayName = (firstName && lastName) ? `${firstName} ${lastName}` : username;
```
Comment (hmhealey) [inline app/components/member_list/index.js]: Same thing with the display name as above

File: service/selectors/entities/users.js [hunk 4]
```diff
@@ -19,6 +33,46 @@ export const getCurrentUser = createSelector(
...
+        currentChannelProfileSet.forEach((p) => {
+            currentProfiles.push({
+                ...profiles[p],
+                status: statuses[p]
+            });
+        });
+
+        // We could get rid of this if server side sorting is a possibility
+        const sortedCurrentProfiles = currentProfiles.sort((a, b) => {
+            const nameA = a.first ? a.first : a.username;
```
Comment (hmhealey) [inline service/selectors/entities/users.js]: We'll need the display name here as well. Also, you have the wrong field name here (`first` should be `first_name`)

**Comment by jarredwitt** [comment]
@hmhealey ready for you

File: app/components/member_list/index.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        onRowPress: PropTypes.func,
+        onListEndReached: PropTypes.func,
+        onListEndReachedThreshold: PropTypes.number,
+        sections: PropTypes.bool,
+        preferences: PropTypes.object
+    }
+
+    static defaultProps = {
+        onListEndReached: () => true,
+        onListEndThreshold: 100,
```
Comment (enahum) [inline app/components/member_list/index.js]: Why is the default 100? it seems like a big number for a default value, maybe 10?

**Comment by jarredwitt** [comment]
@enahum ready for you

**Pull request merged**